onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -649,17 +647,43 @@ struct ProviderHostImpl : ProviderHost {
       elemType = ONNX_NAMESPACE::TensorProto_DataType_UINT4;
     } else if (data_type->s() == "int4") {
       elemType = ONNX_NAMESPACE::TensorProto_DataType_INT4;
-    } else {
-      return;
     }
-    ONNX_NAMESPACE::updateOutputElemType(ctx, 0, elemType);
-    if (shape != nullptr) {
-      for (auto i = 0; i < shape->ints_size(); ++i) {
-        ONNX_NAMESPACE::getOutputShape(ctx, 0, ONNX_NAMESPACE::TypeProto::kTensorType)->add_dim()->set_dim_value(shape->ints(i));
+    return elemType;
+  }
+
+  static void xir_shape_infer(ONNX_NAMESPACE::InferenceContext& ctx) {
+    auto num_output = ctx.getNumOutputs();
+    if (num_output == 1) {
+      auto* shape = ctx.getAttribute("shape");
+      auto* data_type = ctx.getAttribute("data_type");
+      if (data_type == nullptr) {
+        std::cerr << "Custom op is missing `data_type` attr." << std::endl;
+        return;
+      }
+      int32_t elemType = convert_elem_type(data_type);
+      ONNX_NAMESPACE::updateOutputElemType(ctx, 0, elemType);
+      if (shape != nullptr) {
+        for (auto i = 0; i < shape->ints_size(); ++i) {
+          ONNX_NAMESPACE::getOutputShape(ctx, 0, ONNX_NAMESPACE::TypeProto::kTensorType)->add_dim()->set_dim_value(shape->ints(i));
+        }
+      } else {
+        // set scalar type.
+        ONNX_NAMESPACE::getOutputShape(ctx, 0, ONNX_NAMESPACE::TypeProto::kTensorType)->clear_dim();
       }
     } else {
-      // set scalar type.
-      ONNX_NAMESPACE::getOutputShape(ctx, 0, ONNX_NAMESPACE::TypeProto::kTensorType)->clear_dim();
+      for (auto idx = 0u; idx < num_output; idx++) {
+        auto* shape = ctx.getAttribute("shape_" + idx);
```

Comment:
idx is int not string. it causes the build error:
/mnt/vss/_work/1/s/onnxruntime/core/session/provider_bridge_ort.cc:676:49: error: adding 'unsigned int' to a string does not append to the string [-Werror,-Wstring-plus-int]
        auto* shape = ctx.getAttribute("shape_" + idx);
                                       ~~~~~~~~~^~~~~
/mnt/vss/_work/1/s/onnxruntime/core/session/provider_bridge_ort.cc:676:49: note: use array indexing to silence this warning
        auto* shape = ctx.getAttribute("shape_" + idx);
                                                ^
                                       &        [    ]
/mnt/vss/_work/1/s/onnxruntime/core/session/provider_bridge_ort.cc:677:57: error: adding 'unsigned int' to a string does not append to the string [-Werror,-Wstring-plus-int]
        auto* data_type = ctx.getAttribute("data_type_" + idx);
                                           ~~~~~~~~~~~~~^~~~~
/mnt/vss/_work/1/s/onnxruntime/core/session/provider_bridge_ort.cc:677:57: note: use array indexing to silence this warning
        auto* data_type = ctx.getAttribute("data_type_" + idx);
                                                        ^
                                           &            [    ]
2 errors generated.

Reply:
Thanks  @HectorSVC ， this build error has been fixed.